scheduler: "invalid memory address or nil pointer dereference"

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!
-->


**What happened**:
First I create one deployment and one propagationpolicy, it seems to be normal.
```
kubectl create -f samples/nginx/deployment.yaml
kubectl create -f samples/nginx/propagationpolicy.yaml
```

Then I delete the deployment and recreate it, the scheduler seems to be crashed.
```
kubectl delete -f samples/nginx/deployment.yaml
kubectl create -f samples/nginx/deployment.yaml
```

When I delete the propagationpolicy, the scheduler runs normally.
```
kubectl delete -f samples/nginx/propagationpolicy.yaml
```

**What you expected to happen**:
I expected the deployment recreation is normal.

**How to reproduce it (as minimally and precisely as possible)**:
Refer to **What happened**.

**Anything else we need to know?**:
The scheduler's log:
```
[root@11-167-160-68 /data/garryfang/karmada]# kubectl logs karmada-scheduler-6845cd76c9-bl99t -n karmada-system
I0515 11:55:54.284259       1 scheduler.go:162] Starting karmada scheduler
I0515 11:55:54.384747       1 scheduler.go:367] Don't need to schedule binding(lynntest/pg-test-podgroup)
I0515 11:55:54.384822       1 scheduler.go:367] Don't need to schedule binding(lynntest/web-statefulset)
I0515 11:55:54.384866       1 scheduler.go:367] Don't need to schedule binding(lynntest/nginx-service)
E0515 11:55:54.385028       1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 229 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x18d09c0, 0x2832550)
	/root/go/src/github.com/karmada-io/karmada/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:74 +0x95
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/root/go/src/github.com/karmada-io/karmada/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:48 +0x89
panic(0x18d09c0, 0x2832550)
	/usr/local/go/src/runtime/panic.go:969 +0x175
github.com/karmada-io/karmada/pkg/scheduler/framework.PluginToResult.Merge(0xc000b327e0, 0x1d1cf40)
	/root/go/src/github.com/karmada-io/karmada/pkg/scheduler/framework/interface.go:86 +0xf2
github.com/karmada-io/karmada/pkg/scheduler/core.(*genericScheduler).findClustersThatFit(0xc0009572f0, 0x1d1cf40, 0xc00012e018, 0x1ce54c0, 0xc0009572c0, 0xc000330b68, 0xc000b2f3e0, 0x19, 0x413f1d, 0xc000b16920, ...)
	/root/go/src/github.com/karmada-io/karmada/pkg/scheduler/core/generic_scheduler.go:85 +0x113
github.com/karmada-io/karmada/pkg/scheduler/core.(*genericScheduler).Schedule(0xc0009572f0, 0x1d1cf40, 0xc00012e018, 0xc000330b68, 0x1, 0xc0004f5ba8, 0x16efc1d, 0xc000954f20, 0xc000b16c20)
	/root/go/src/github.com/karmada-io/karmada/pkg/scheduler/core/generic_scheduler.go:54 +0xaa
github.com/karmada-io/karmada/pkg/scheduler.(*Scheduler).scheduleResourceBinding(0xc00020a800, 0xc000385310, 0xc000330a30, 0xc000330a30, 0x0)
	/root/go/src/github.com/karmada-io/karmada/pkg/scheduler/scheduler.go:417 +0x88
github.com/karmada-io/karmada/pkg/scheduler.(*Scheduler).scheduleOne(0xc00020a800, 0xc0004cf3a0, 0x18, 0x0, 0x0)
	/root/go/src/github.com/karmada-io/karmada/pkg/scheduler/scheduler.go:413 +0x7aa
github.com/karmada-io/karmada/pkg/scheduler.(*Scheduler).scheduleNext(0xc00020a800, 0x203000)
	/root/go/src/github.com/karmada-io/karmada/pkg/scheduler/scheduler.go:356 +0x332
github.com/karmada-io/karmada/pkg/scheduler.(*Scheduler).worker(0xc00020a800)
	/root/go/src/github.com/karmada-io/karmada/pkg/scheduler/scheduler.go:248 +0x2b
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc000b15130)
	/root/go/src/github.com/karmada-io/karmada/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155 +0x5f
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000b15130, 0x1cdad20, 0xc000b1d620, 0x1, 0xc0003fea80)
	/root/go/src/github.com/karmada-io/karmada/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156 +0xad
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000b15130, 0x3b9aca00, 0x0, 0x1bb7901, 0xc0003fea80)
	/root/go/src/github.com/karmada-io/karmada/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x98
k8s.io/apimachinery/pkg/util/wait.Until(0xc000b15130, 0x3b9aca00, 0xc0003fea80)
	/root/go/src/github.com/karmada-io/karmada/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:90 +0x4d
created by github.com/karmada-io/karmada/pkg/scheduler.(*Scheduler).Run
	/root/go/src/github.com/karmada-io/karmada/pkg/scheduler/scheduler.go:169 +0x245
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x16fba92]

goroutine 229 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/root/go/src/github.com/karmada-io/karmada/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:55 +0x10c
panic(0x18d09c0, 0x2832550)
	/usr/local/go/src/runtime/panic.go:969 +0x175
github.com/karmada-io/karmada/pkg/scheduler/framework.PluginToResult.Merge(0xc000b327e0, 0x1d1cf40)
	/root/go/src/github.com/karmada-io/karmada/pkg/scheduler/framework/interface.go:86 +0xf2
github.com/karmada-io/karmada/pkg/scheduler/core.(*genericScheduler).findClustersThatFit(0xc0009572f0, 0x1d1cf40, 0xc00012e018, 0x1ce54c0, 0xc0009572c0, 0xc000330b68, 0xc000b2f3e0, 0x19, 0x413f1d, 0xc000b16920, ...)
	/root/go/src/github.com/karmada-io/karmada/pkg/scheduler/core/generic_scheduler.go:85 +0x113
github.com/karmada-io/karmada/pkg/scheduler/core.(*genericScheduler).Schedule(0xc0009572f0, 0x1d1cf40, 0xc00012e018, 0xc000330b68, 0x1, 0xc0004f5ba8, 0x16efc1d, 0xc000954f20, 0xc000b16c20)
	/root/go/src/github.com/karmada-io/karmada/pkg/scheduler/core/generic_scheduler.go:54 +0xaa
github.com/karmada-io/karmada/pkg/scheduler.(*Scheduler).scheduleResourceBinding(0xc00020a800, 0xc000385310, 0xc000330a30, 0xc000330a30, 0x0)
	/root/go/src/github.com/karmada-io/karmada/pkg/scheduler/scheduler.go:417 +0x88
github.com/karmada-io/karmada/pkg/scheduler.(*Scheduler).scheduleOne(0xc00020a800, 0xc0004cf3a0, 0x18, 0x0, 0x0)
	/root/go/src/github.com/karmada-io/karmada/pkg/scheduler/scheduler.go:413 +0x7aa
github.com/karmada-io/karmada/pkg/scheduler.(*Scheduler).scheduleNext(0xc00020a800, 0x203000)
	/root/go/src/github.com/karmada-io/karmada/pkg/scheduler/scheduler.go:356 +0x332
github.com/karmada-io/karmada/pkg/scheduler.(*Scheduler).worker(0xc00020a800)
	/root/go/src/github.com/karmada-io/karmada/pkg/scheduler/scheduler.go:248 +0x2b
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc000b15130)
	/root/go/src/github.com/karmada-io/karmada/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155 +0x5f
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000b15130, 0x1cdad20, 0xc000b1d620, 0x1, 0xc0003fea80)
	/root/go/src/github.com/karmada-io/karmada/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156 +0xad
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000b15130, 0x3b9aca00, 0x0, 0x1bb7901, 0xc0003fea80)
	/root/go/src/github.com/karmada-io/karmada/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x98
k8s.io/apimachinery/pkg/util/wait.Until(0xc000b15130, 0x3b9aca00, 0xc0003fea80)
	/root/go/src/github.com/karmada-io/karmada/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:90 +0x4d
created by github.com/karmada-io/karmada/pkg/scheduler.(*Scheduler).Run
	/root/go/src/github.com/karmada-io/karmada/pkg/scheduler/scheduler.go:169 +0x245
```

**Environment**:
- Karmada version:  latest
- Others:

--- a/pkg/scheduler/framework/plugins/clusteraffinity/cluster_affinity.go
+++ b/pkg/scheduler/framework/plugins/clusteraffinity/cluster_affinity.go
@@ -41,7 +41,7 @@ func (p *ClusterAffinity) Filter(ctx context.Context, placement *v1alpha1.Placem
 	}
 
 	// If no clusters specified and it is not excluded, mark it matched
-	return nil
+	return framework.NewResult(framework.Success)
 }
 
 // Score calculates the score on the candidate cluster.

